src/mame/rm/rm_mq2.cpp

```diff
@@ -135,9 +136,13 @@ void rmMQ2_device::input_rts(int state)
 {
 	if (started())
 	{
-	m_sio->dcdb_w(state);
-	// NMI generated to reset baud rate to 9600
-	m_maincpu->set_input_line(INPUT_LINE_NMI, state ? ASSERT_LINE : CLEAR_LINE);
+		m_sio->dcdb_w(state);
+		// give firmware ~50ms to finish initialisation before allowing first NMI
+		if (m_maincpu->total_cycles() > 200'000)
```

Comment:
On hardware, is there some kind of handshake or something that makes this work, or is it more of a magical timing thing?

Reply:
It seems to be a magical timing thing, but I suppose with real hardware it would very unusual to start both CPUs at exactly the same time - there would naturally be at least a few seconds gap between pressing one power switch and then the other!